tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -518,6 +519,86 @@ public void changing(LocationEvent event) {
 	for (int i = 0; i < 100; i++) browser.removeLocationListener(listener);
 }
 
+@Test
+public void test_isLocationForCustomText_setUrlAfterDisposedThrowsSwtException() {
+	Browser testBrowser = createBrowser(shell, SWT.NONE);
+	testBrowser.dispose();
+	assertThrows(SWTException.class, () -> testBrowser.isLocationForCustomText("about:blank"));
+}
+
+@Test
+public void test_isLocationForCustomText_isSetUrlNotCustomTextUrlAfterSetText() {
+	assumeFalse("behavior is not (yet) supported by Edge browser", isEdge);	// Edge doesn't fire a changed event if the URL is the same as the previous location.
+	URI url = getValidUrl();
+	AtomicBoolean locationChanged = new AtomicBoolean(false);
+	browser.addLocationListener(changedAdapter(event -> {
+			locationChanged.set(true);
+	}));
+
+	browser.setText("Custom text");
+	assertTrue("Time Out: The Browser didn't navigate to the URL", waitForPassCondition(locationChanged::get));
+	locationChanged.set(false);
+	browser.setUrl(url.toASCIIString());
+	shell.open();
```

Comment:
Just wondering: it does not seem to matter when (or even it at all) the shell is opened. Still is seems weird that it is opened after all operations have already been executed. Maybe the shell should be opened at the beginning of the test to avoid confusion (at least it should be done at a consistent point in time throughout the tests)? Or is there a reason for doing it here?

Reply:
It doesnt matter at all haha. Te tests even work without opening the shell. It's just for the visibility's sake but I think, if we don't open the shell, the tests run faster. Also we don't have those blinking windows on the monitor for the time the tests are running.

Reply:
Removed the shell.open calls from new tests